### PR TITLE
Add .NET core test runner

### PR DESCRIPTION
**Pull request by nirinchev**
Supersedes #1475

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -469,6 +481,19 @@ def AndroidTest(stashName) {
...
 }
 
+def NetCoreTest(String node, String stashName) {
+  return {
+    nodeWithCleanup(node) {
+      getArchive()
+      unstash stashName
+
+      dir ("Tests/Tests.NetCore/bin/${configuration}/netcoreapp1.1") {
+        sh 'dotnet Tests.NetCore.dll'
```
Comment (fealebenpae) [inline Jenkinsfile]: You might also want to report the result to Jenkins.

Comment (nirinchev) [inline Jenkinsfile]: Yes, I'm taking baby steps - first make sure it runs, next - transform results and report them :)

File: Tests/Tests.NetCore/Tests.NetCore.csproj [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\Realm\Realm\Realm.csproj" />
+    <ProjectReference Include="..\..\Realm\Realm.Sync\Realm.Sync.csproj" Condition="'$(RealmNoSync)' != 'true'" />
+  </ItemGroup>
+
+  <Import Project="..\Tests.Shared\Tests.Shared.projitems" Label="Shared" />
+  <Import Project="..\Tests.Sync.Shared\Tests.Sync.Shared.projitems" Label="Shared" Condition="'$(RealmNoSync)' != 'true'" />
+  
+  <Import Project="..\..\packages\Fody.2.0.6\build\netstandard1.4\Fody.targets" />
```
Comment (nirinchev) [inline Tests/Tests.NetCore/Tests.NetCore.csproj]: Look into why that's needed

File: Tests/Tests.Shared/NotificationTests.cs [hunk 3]
```diff
@@ -32,8 +32,6 @@ namespace Tests.Database
     [TestFixture, Preserve(AllMembers = true)]
     public class NotificationTests
     {
-        private const int MillisecondsToWaitForCollectionNotification = 50;
```
Comment (nirinchev) [inline Tests/Tests.Shared/NotificationTests.cs]: We should investigate if there's a real bug with autorefreshing realms - on .NET Framework apps, this seems to work.

File: Jenkinsfile [hunk 4]
```diff
@@ -406,10 +405,33 @@ stage('Build with sync') {
...
 }
 
+stage ('Build .NET Core with sync') {
+  nodeWithCleanup('windows') {
+    getArchive()
+    unstash 'macos-wrappers-sync'
+    unstash 'linux-wrappers-sync'
+    unstash 'tools-weaver'
+
+    bat 'dotnet restore Tests/Tests.NetCore/Tests.NetCore.csproj'
```
Comment (fealebenpae) [inline Jenkinsfile]: This shouldn't be necessary if you're calling the Restore target in msbuild

File: Jenkinsfile [hunk 5]
```diff
@@ -748,7 +770,7 @@ def transform(String original, String transform) {
     """
   } else {
     bat """
-      powershell \"\$xml = Resolve-Path ${original};\$output = Join-Path (\$pwd) temp.xml;\$xslt = New-Object System.Xml.Xsl.XslCompiledTransform;\$xslt.Load(\"${transform}\");\$xslt.Transform(\$xml, \$output);\"
+      powershell \"\$xml = Resolve-Path ${original};\$output = Join-Path (\$pwd) temp.xml;\$xslt = New-Object System.Xml.Xsl.XslCompiledTransform;\$xslt.Load(\\\"${transform}\\\");\$xslt.Transform(\$xml, \$output);\"
```
Comment (fealebenpae) [inline Jenkinsfile]: Can't you just do that in Groovy? 😛

Comment (nirinchev) [inline Jenkinsfile]: Hm... I wasn't able to find how, but I'm not a groovy expert, so maybe I wasn't looking at the right place :)

@emanuelez is there an easy way to apply an xslt transform in groovy?

File: Jenkinsfile (hunk 4 above)
Comment (nirinchev) [inline Jenkinsfile]: Not sure why, but it is. `dotnet restore` creates a `.targets` file that imports the Fody targets, whereas I wasn't able to see such created by `msbuild /t:Restore`. Consequently, if we don't run `dotnet restore`, the weaver is not run.

File: Jenkinsfile (hunk 5 above)
Comment (emanuelez) [inline Jenkinsfile]: There should really be as little logic as possible in Groovy so I think this approach is fine. What are you transforming by the way? If it's unit test reports I'm sure the xunit plugin could help.

Comment (nirinchev) [inline Jenkinsfile]: Yes - it's a test result. I'm not familiar with xunit, but so far we've been transforming all our test results, so perhaps there's a reason which @fealebenpae knows :)

Comment (emanuelez) [inline Jenkinsfile]: ok :) here's the wiki page so you can check whether your test framework is supported. https://wiki.jenkins.io/display/JENKINS/xUnit+Plugin

Comment (nirinchev) [inline Jenkinsfile]: Ah... I [guess it's not](https://issues.jenkins-ci.org/browse/JENKINS-33450), which is why we're not using it.

Comment (fealebenpae) [inline Jenkinsfile]: The xunit plugin also prematurely sets the build result to successful, if I remember correctly.

File: Tests/Tests.Shared/NotificationTests.cs [hunk 6]
```diff
@@ -645,7 +645,13 @@ public void TestCollectionChangedAdapter(int[] initial, NotifyCollectionChangedA
...
         public void WhenSynchronizationContextExists_ShouldAutoRefresh()
         {
-            AsyncContext.Run(async () =>
+#if NETCOREAPP1_1
+            if (TestHelpers.IsMacOS)
+            {
+                Assert.Ignore("Seems like there's a bug with autorefreshing on .NET Core on macOS");
+            }
+#endif
+			AsyncContext.Run(async () =>
```
Comment (fealebenpae) [inline Tests/Tests.Shared/NotificationTests.cs]: indentation

**Pull request merged**